metaflow/plugins/kfp/kfp.py

```diff
@@ -28,35 +29,166 @@ def get_ordered_steps(graph):
+    S3_BUCKET = "s3://workspace-zillow-analytics-stage/aip/metaflow"
+    S3_AWS_ARN = "arn:aws:iam::170606514770:role/dev-zestimate-role"
+
+    define_s3_env_vars = 'export METAFLOW_DATASTORE_SYSROOT_S3="{}" && export METAFLOW_AWS_ARN="{}" '.format(S3_BUCKET,
+                                                                                                             S3_AWS_ARN)
+    define_username = 'export USERNAME="kfp-user"'
+    python_cmd = "python helloworld.py --datastore s3 --datastore-root {0} step {1} --run-id {2} " \
+                 "--task-id {3} --input-paths {2}/{4}/{5}".format(ds_root, step_name, run_id,
+                                                                  task_id, prev_step_name, prev_task_id)
+    final_run_cmd = f'{define_username} && {define_s3_env_vars} && {python_cmd}'
+
+    print("RUNNING COMMAND: ", final_run_cmd)
+    proc = subprocess.run(final_run_cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE, universal_newlines=True)
+    proc_output = proc.stdout
+    proc_error = proc.stderr
+
+    step_output = namedtuple('StepOutput',
+                             ['ds_root', 'run_id', 'next_step', 'next_task_id', 'current_step', 'current_task_id'])
+
+    # END is the final step and no outputs need to be returned
+    if step_name.lower() == 'end':
+        print("_______________ FLOW RUN COMPLETE ________________")
+        return step_output('None', 'None', 'None', 'None', 'None', 'None')
+
+    # TODO: Check why MF echo statements are getting redirected to stderr
+    if len(proc_error) > 1:
+        print("_______________STDERR:_____________________________")
+        print(proc_error)
+
+    if len(proc_output) > 1:
+        print("_______________STDOUT:_____________________________")
+        print(proc_output)
+        # Read in the outputs (in the penultimate line) containing args needed for the next step.
+        # Note: Output format is: ['ds_root', 'run_id', 'next_step', 'next_task_id', 'current_step', 'current_task_id']
+        outputs = (proc_output.split("\n")[-2]).split()
+        print(step_output(outputs[0], outputs[1], outputs[2], outputs[3], outputs[4], outputs[5]))
+    else:
+        raise RuntimeWarning("This step did not generate the correct args for next step to run. This might disrupt "
+                             "the workflow")
+
+    # TODO: Metadata needed for client API to run needs to be persisted outside before this
+    print("--------------- RUNNING: CLEANUP OF TEMP DIR----------")
+    subprocess.call(["rm -r /opt/zillow/.metaflow"], shell=True)
+
+    print("_______________ Done _________________________________")
+    return step_output(outputs[0], outputs[1], outputs[2], outputs[3], outputs[4], outputs[5])
+
+def pre_start_op_func(code_url)  -> NamedTuple('StepOutput', [('ds_root', str), ('run_id', str), ('next_step', str), ('next_task_id', str), ('current_step', str), ('current_task_id', str)]):
```

Comment:
declare `StepOutput` as a named tuple outside of this function
example:
```python
DateKeyPaths = NamedTuple("date_key_paths", [("train_path", str), ("out_path", str)])
```

Reply:
Done